rmgpy/data/thermo.py

```diff
@@ -1600,18 +1604,42 @@ def get_thermo_data_for_surface_species(self, species):
                         adsorbed_atoms[1].increment_lone_pairs()
                     else:
                         adsorbed_atoms[0].increment_lone_pairs()
+                    dummy_molecules.append(dummy_molecule.copy(deep=True))
 
-        dummy_molecule.update_connectivity_values()
-        dummy_molecule.update()
+        for dummy_molecule in dummy_molecules[:]:
+            try:
+                dummy_molecule.update_connectivity_values()
+                dummy_molecule.update()
+            except:
+                dummy_molecules.remove(dummy_molecule)
+                logging.debug(f"Removing {dummy_molecule} from possible structure list:\n{dummy_molecule.to_adjacency_list()}")
+            else:
+                logging.debug("After removing from surface:\n" + dummy_molecule.to_adjacency_list())
 
-        logging.debug("After removing from surface:\n" + dummy_molecule.to_adjacency_list())
+        if len(dummy_molecules) == 0:
+            raise RuntimeError(f"Cannot get thermo for gas-phase molecule. No valid dummy molecules from original molecule:\n{molecule.to_adjacency_list()}")
+
+        
+        # if len(molecule) > 1, it will assume all resonance structures have already been generated when it tries to generate them, so evaluate each configuration separately and pick the lowest energy one by H298 value
+        gas_phase_species = []
+        for dummy_molecule in dummy_molecules:
+            dummy_species = Species()
+            dummy_species.molecule = [dummy_molecule]
+            dummy_species.generate_resonance_structures()
+            dummy_species.thermo = self.get_thermo_data(dummy_species)
+            gas_phase_species.append(dummy_species)
+            
+        # define the comparison function to find the lowest energy
+        def lowest_energy(species):
+            if hasattr(species.thermo, 'H298'):
+                return species.thermo.H298.value
```

Comment:
Can you please change this to `.value_si`?  `get_enthalpy` will return SI units (J/mol), but the ThermoData H298 may have different units.

Reply:
Good catch!  I almost merged this, thinking it had already been reviewed enough. But following the best practice of rigorous peer review saves the day, again.
Thanks @davidfarinajr for doing it carefully.

Reply:
```suggestion
                return species.thermo.H298.value_si
```